lightning/src/ln/onion_utils.rs

```diff
@@ -987,7 +987,8 @@ where
 	.expect("Route we used spontaneously grew invalid keys in the middle of it?");
 
 	// Handle packed channel/node updates for passing back for the route handler
-	for (route_hop_idx, (route_hop_option, shared_secret)) in onion_keys.into_iter().enumerate() {
+	let mut iterator = onion_keys.into_iter().peekable();
```

Comment:
I am not sure if this commit is really the best idea. I've been (incompletely) rebasing attributable failures on top, and there I need that route_hop_idx again to index into the hmacs and payloads.

https://github.com/lightningdevkit/rust-lightning/compare/main...joostjager:rust-lightning:attr-errs-on-trampoline

Reply:
Yes that works